src/arrayMethodSort.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let comparator = compareFunction;
+
+    if (comparator === undefined) {
+      comparator = function(a, b) {
+        if (String(a) < String(b)) {
+          return -1;
+        }
+
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        return 0;
+      };
+    }
+
+    let value;
+
+    for (let i = this.length - 1; i > 0; i--) {
```

Comment:
use do while, your solution very difficult. When you don't need make change you do extra iterations